programs/clearing_house/src/controller/orders.rs

```diff
@@ -1013,7 +1017,19 @@ pub fn fulfill_order_with_amm(
             order_post_only,
         )?;
 
+    let market_postion_unsettled_pnl_delta = cast_to_i128(user_fee)?
+        .checked_sub(cast_to_i128(filler_reward)?)
+        .ok_or_else(math_error!())?;
+
+    let mut pnl = update_user_and_market_position(
```

Comment:
i really dont like that we update part of the amm in `swap_base_asset_position_delta` and then we update the market position here.

pretty sure we did this because we need to get the `fee_to_market` to apply to the market position.

In order to keep consistency with how we update the position _and then_ apply the fee, maybe we should break `update_user_and_market_position` into two functions, one that updates the positions given the swap and then another one that applies the fee. will take a stab at this